iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ThreeWindingsTransformerImpl.java

```diff
@@ -382,6 +383,27 @@ public Terminal getTerminal(ThreeSides side) {
         };
     }
 
+    @Override
+    public Terminal getTerminal(String voltageLevelId) {
+        Objects.requireNonNull(voltageLevelId);
+        boolean side1 = getLeg1().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
+        boolean side2 = getLeg2().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
+        boolean side3 = getLeg3().getTerminal().getVoltageLevel().getId().equals(voltageLevelId);
```

Comment:
`Terminal.getVoltageLevel()` can return null.
It would be safer to use this method to compute the "side..." boolean:
```java
    private boolean isLegOnVoltageLevel(ThreeWindingsTransformer.Leg leg, String voltageLevelId) {
        return Optional.ofNullable(leg.getTerminal().getVoltageLevel())
                .map(vl -> voltageLevelId.equals(vl.getId()))
                .orElse(Boolean.FALSE);
    }
```